docs/design/rosbag2_playback_time.md

```diff
@@ -187,6 +187,53 @@ No special allowance needs to be made in the `Player` - time will be provided at
 
 The `Player` must register a `JumpHandler` with the `Clock` - so that when a jump occurs, the current message playback queue can be invalidated and re-enqueued according to the new starting time.
 
+## Time Synchronization and Conversion
+
+When implementing this time control, there are two separate time streams to consider:
+1. "ROS time": this is the time that is related to the messages in the bag - this time may speed up, slow down, pause, or jump based on rosbag playback controls (or external Time Sources)
+2. "Steady time": this time always moves forward, and always moves at the same rate; it is the real time that the user watching a playback experiences
+
+To implement `now() -> ROSTime` and `sleep_until(ROSTime until)`, we need to perform arbitrary conversion between "ROS Time" and "Steady Time".
+
+We use the following diagrams to visualize the derivation for these conversion equations.
+Steady Time is marked as timeline `S`, and ROS Time as timeline `R`.
+Any `R_n` and `S_n` are matched time points, they coincide with each other: at Steady Time `S_n`, `now()` will return `R_n`.
+Likewise, when `sleep_until(R_n)` is called, the `PlayerClock` which uses steady time for sleeping, will sleep until `S_n`.
+
+In this diagram you can see `rate` changes, `pause`, and `jump`, with arrows showing matched points.
+The bag we are playing starts at R0 and goes on forward in time (we do not show the end of the bag).
+The user is performing actions in `S`, the timeline that they experience.
+
+The lines and numbered references indicate "events" where a time control operation has been performed, such as changing the rate, pausing, or jumping.
+
+Events in the first timeline "Rate Change":
+1. At `S_0`: start playback at rate `0.5`
+1. At `S_1`: change rate to `2.0`
+1. At `S_2`: change rate to `1.0`
```

Comment:
@emersonknapp It seems you misinterpreting playback rate. \
Playback rate is sort of multiplication factor for the speed and should be quite opposite to how you pictured it. \
i.e when playback rate 0.5 the messages should be played 1/2 of the original speed with 2x longer interval. \
At least this is common convention for media players.

I think it should be as:
```txt
Events in the first timeline "Rate Change":
At S_0: start playback at rate 2.0
At S_1: change rate to 0.5
At S_2: change rate to 1.0
```

Reply:
Ok got it. This ROS timeline kind of confusing me since I am expected to see time intervals for actual playback on it.

Reply:
I'd like to clarify that when I say "actual playback" I am referring to the time that I experience as a user - this corresponds to the Steady Time. When you say "actual playback", it seem that you mean ROS Timestamps, something that can change (according to the observer).  Clearly this "actual" term is ambiguous and will not help this discussion.

I would like to emphasize that you are seeing "acual intervals" for both times - neither timeline is scaled in any way, a distance along one timeline is the same as the distance in another. For example 100mm of diagram corresponds to 1minute of time, for either `R` or `S` - the only difference is how fast these move in relation to each other. While `rate = 2.0`, a 100mm segment of `R`  will match up with  a 50mm segment of `S` (or from the other frame of reference, if `S` moves forward 100mm, `R` will move 200mm)